libs/metrics/src/lib.rs

```diff
@@ -32,13 +33,27 @@ macro_rules! register_uint_gauge_vec {
     }};
 }
 
+/// Special internal registry, to collect metrics independently from the default registry.
+/// Was introduced to fix deadlock with lazy registration of metrics in the default registry.
+static INTERNAL_REGISTRY: Lazy<Registry> = Lazy::new(Registry::new);
+
+/// Register a collector in the internal registry. MUST be called before the first call to `gather()`.
```

Comment:
this can be hardened by using OnceCell instead of Lazy

Reply:
I'm not sure I follow, OnceCell is just a more explicit version of Lazy, which doesn't accept closure in constructor. I'm going to merge this PR, feel free to create another one to replace Lazy with OnceCell.